index.js

```diff
@@ -74,15 +74,25 @@ module.exports = function (processors, options) {
     }
 
     function handleError (error) {
-      var errorOptions = { fileName: file.path, showStack: true }
-      if (error.name === 'CssSyntaxError') {
-        error = error.message + '\n\n' + error.showSourceCode() + '\n'
-        errorOptions.showStack = false
+      var errorOptions = { fileName: file.path }
+      if (error instanceof Object) {
+        errorOptions.error = error
+        if (error.name === 'CssSyntaxError') {
+          var input = error.input || {};
+          errorOptions.fileName = error.file || input.file || file.path
```

Comment:
Shouldn't the `fileName` be the same as `file.path`, because the error was triggered by this file? Can you elaborate on this logic?